src/Services/Catalog/Catalog.Infrastructure/Repositories/CategoryRepository.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Catalog.Domain.Abstractions;
+using Catalog.Domain.Entities;
+using Microsoft.EntityFrameworkCore;
+using MongoDB.Driver;
+
+namespace Catalog.Infrastructure.Repositories;
+
+public class CategoryRepository : ICategoryRepository
+{
+    private readonly IMongoCollection<Category> _categories;
+    private readonly IMongoCollection<Product> _products;
+
+    private readonly InsertOneOptions _insertOneOptions;
+    private readonly EstimatedDocumentCountOptions _estimatedDocumentCountOptions;
+    private readonly DeleteOptions _deleteOneOptions;
+
+    public CategoryRepository(IMongoCollection<Category> categories, IMongoCollection<Product> products)
```

Comment:
you should define abstract class BaseRepository, where you should implement all of methods you described in IBaseRepository interface. and in your repositories you should inherit from BaseRepository and ICategoryRepository